File: solutions/_374_guess_number_higher_or_lower/solution.js
class GuessGame {
  constructor(pick) {
    this.pick = pick;
  }

  guess(num) {
    if (num > this.pick) {
      return -1;
    } else if (num < this.pick) {
      return 1;
    } else {
      return 0;
    }
  }


  /**
   * @param {number} n
   * @return {number}
   */
  guessNumber(n) {
    let lo = 1;
    let hi = n;

    while (true) {
      let mid = lo + (hi - lo) / 2;
      let myGuess = this.guess(mid);
      if (myGuess == 1)
        lo = mid + 1;
      else if (myGuess == -1)
        hi = mid - 1;
      else
        return mid;
    }
  }
}

module.exports = { GuessGame };

/*
https://leetcode.com/problems/guess-number-higher-or-lower/
*/

File: solutions/_374_guess_number_higher_or_lower/solution.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:


class Solution:
    def __init__(self, pick: int):
        self.pick = pick

    def guess(self, num: int) -> int:
        if num > self.pick:
            return -1
        elif num < self.pick:
            return 1
        else:
            return 0

    def guessNumber(self, n: int) -> int:
        lo = 1
        hi = n

        while True:
            mid = lo + (hi - lo) // 2
            myGuess = self.guess(mid)
            if myGuess == 1:
                lo = mid + 1
            elif myGuess == -1:
                hi = mid - 1
            else:
                return mid


# https://leetcode.com/problems/guess-number-higher-or-lower/
